backend/services/foodService.ts

```diff
@@ -0,0 +1,52 @@
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+export async function saveFoodData(
+  userId: string,
+  imageData: string,
+  foodData: any
+) {
+  try {
+    // Early return if no food data or food items
+    if (!foodData || !foodData.foodItems || foodData.foodItems.length === 0) {
+      console.log("No food items detected, skipping database save");
+      return null;
+    }
+
+    // First, ensure user exists (create if not)
+    const user = await prisma.user.upsert({
+      where: { clerkId: userId },
+      update: {},
+      create: {
+        clerkId: userId,
+      },
+    });
+
+    // Create the image with required fields
+    const image = await prisma.image.create({
+      data: {
+        userId: user.id,
+        imageUrl: "https://example.com/image.jpg", // for the future we can store imageData in the cloud
+        foodItems: {
+          create:
+            foodData.foodItems?.map((item: any) => ({
+              name: item.name,
```

Comment:
We shouldn't have a ? next to foodItems here. The case where foodItems is undefined should be handled closer to the AI service.

Reply:
fixed